src/layout/HorizontalPanAndZoomView.js

```diff
@@ -21,11 +21,16 @@ import {
   MOVE_WHEEL_DELTA_THRESHOLD,
 } from '../canvas/constants'; // TODO: Remove external dependency
 
-type HorizontalPanAndZoomState = {|
+type HorizontalPanAndZoomState = $ReadOnly<{|
```

Comment:
Curious, what does `$ReadOnly` do here?

Reply:
It prevents people from reassigning the values of the object. So you can't do e.g. `state.offsetX = 234`